harmony_regridding_service/regridding_service.py

```diff
@@ -623,11 +592,15 @@ def _copy_var_without_metadata(
     """
     var = PurePath(variable_name)
     s_var = _get_variable(source_ds, variable_name)
+
+    # Create target variable
     t_group = target_ds.createGroup(var.parent)
     fill_value = getattr(s_var, '_FillValue', None)
     t_var = t_group.createVariable(
         var.name, s_var.dtype, s_var.dimensions, fill_value=fill_value
     )
+    s_var.set_auto_maskandscale(False)
+    t_var.set_auto_maskandscale(False)
```

Comment:
This stops implicit masking and scaling. This was causing issues when the datatypes were being converted. tl;dr missing was being converted to 1e20 and then casted back to a smaller value, apple silicon turned it to zero, amd64 set it to the fill value (max int64 -2). Now casting and masking are handled explicitly